docs: group app keys into general and daemon sections (#6293)
snapcore/snapcraft: docs/reference/snapcraft-yaml.rst

> .. meta:

Reviewer: The original syntax was correct.
https://docutils.sourceforge.io/docs/ref/rst/directives.html#meta

> .. meta:

Author: ```suggestion
.. meta::

.. kitbash-field:: Project apps

General app keys
~~~~~~~~~~~~~~~~

> These keys apply to app definitions generally, whether the app is run as a command or as a service.

Reviewer: ```suggestion
These keys can be declared for apps whether they run as commands or services.
```
This is slightly more consistent with the language we use surrounding keys in the craft apps.

(My use of "they" is slightly ambiguous, but I don't think it'll cause any confusion.)

.. kitbash-field:: App completer
    :prepend-name: apps.<app-name>

.. kitbash-field:: App slots
    :prepend-name: apps.<app-name>

.. kitbash-field:: App plugs
    :prepend-name: apps.<app-name>

.. kitbash-field:: App aliases
    :prepend-name: apps.<app-name>

.. kitbash-field:: App environment
    :prepend-name: apps.<app-name>

.. kitbash-field:: App command_chain
    :prepend-name: apps.<app-name>

.. kitbash-field:: App sockets
    :prepend-name: apps.<app-name>

> 

Author: I did not use it in services only section as socket setting in the source code is:

- Not ignored for executable apps by code
- Not conditionally applied only when daemon is set

But it makes more sense in a service. So going ahead and fixing it as per code comments indicating suggestive use. Thanks.

.. kitbash-field:: App passthrough
    :prepend-name: apps.<app-name>

.. kitbash-field:: App extensions
    :prepend-name: apps.<app-name>

Service and daemon keys
~~~~~~~~~~~~~~~~~~~~~~~

> These keys apply to apps configured to run as services or daemons. Most are relevant only when ``daemon`` is set.

Reviewer: ```suggestion
These keys can be declared for apps run as services or daemons.
```
> Most are relevant only when ``daemon`` is set.

It looks like this is true for all of the keys but the daemon key itself. Since this is mentioned in every key's description, I think it's safe to drop it here.

.. kitbash-field:: App timer
    :prepend-name: apps.<app-name>

.. _reference-snapcraft-yaml-app-keys-app-daemon:

> 

Reviewer: ```suggestion
```

The following Kitbash directive generates the label `App.daemon`, which should make this redundant.

> .. kitbash-field:: App daemon

Reviewer 2: Since all daemon keys are dependent on this one, let's make this the first in the section.
